src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -3981,6 +5082,115 @@ type TicketType implements Node {
     zones: [String!]
 }
 
+input TimetablePreferencesInput {
+    """
+    When false, real-time updates are considered during the routing.
+    In practice, when this option is set as true, some of the suggestions might not be
+    realistic as the transfers could be invalid due to delays,
+    trips can be cancelled or stops can be skipped.
+    """
+    excludeRealTimeUpdates: Boolean = false
+
+    """
+    When true, departures that have been cancelled ahead of time will be
+    included during the routing. This means that an itinerary can include
+    a cancelled departure while some other alternative that contains no cancellations
+    could be filtered out as the alternative containing a cancellation would normally
+    be better.
+    """
+    includePlannedCancellations: Boolean = false
+
+    """
+    When true, departures that have been cancelled through a real-time feed will be
+    included during the routing. This means that an itinerary can include
+    a cancelled departure while some other alternative that contains no cancellations
+    could be filtered out as the alternative containing a cancellation would normally
+    be better. This option can't be set to true while `includeRealTimeUpdates` is false.
+    """
+    includeRealTimeCancellations: Boolean = false
```

Comment:
Should these have hard-coded default values in the schema? Currently Transmodel API has them. It's possible to configure value for `excludeRealtimeUpdates` (`ignoreRealtimeUpdates`) in default route request.

Reply:
We decided to remove the defaults. I have done the same for number of itineraries as it can be configured as well.